contributors/design-proposals/storage/csi-migration.md

```diff
@@ -35,12 +35,28 @@ lower cost of development as we only have to maintain one version of each
 plugin, as well as ease the transition to CSI when we are able to deprecate the
 internal APIs.
 
+The migration from in-tree plugins to CSI plugins will involve the following phases:
+
+Phase 1: In-tree plugins are probed but the in-tree plugin code is largely skipped
+for operations on persistent volumes in favor of CSI plugins that supersede the
+in-tree plugins. In case a node does not have a CSI plugin (that supersedes a
+given in-tree plugin) installed and configured, the in-tree plugin code
+will be executed for specific operations like attachment/detachment and attachment
+verification of volumes.
+
+Phase 2: In-tree plugins are no longer probed and thus in-tree code is effectively
```

Comment:
can this be done on a plugin by plugin basis or will all in-tree plugins not be probed at the same time? please clarify in text. (I think it should have the option to be done piecemeal)

Reply:
This will be doable in a plugin-by-plugin basis. Will clarify.